### PR TITLE
Implement Workflow

### PR DESCRIPTION
**Pull request by Aman-Pandey-afk**
Added basic directory structure and CI workflow, will add codecov in the same PR but different commit. (After a check for CI result)

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          - '1.7'
+          - 'nightly'
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
+        arch:
+          - x64
+    steps:
+      - uses: actions/checkout@v2
```
Comment (giordano) [inline .github/workflows/CI.yml:30]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      - uses: codecov/codecov-action@v2
+        with:
+          files: lcov.info
+  docs:
+    name: Documentation
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v2
```
Comment (giordano) [inline .github/workflows/CI.yml:48]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: '1'
+      - uses: julia-actions/julia-buildpkg@v1
+      - uses: julia-actions/julia-docdeploy@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: |
+          julia --project=docs -e '
```
Comment (giordano) [inline .github/workflows/CI.yml:57]:
```suggestion
          julia --color=yes --project=docs -e '
```

File: test/runtests.jl [hunk 4]
```diff
@@ -0,0 +1,6 @@
+using Stingray
+using Test
+
+@testset "Stingray.jl" begin
+    @test typeof(2) <: Real
+end
```
Comment (giordano) [inline test/runtests.jl:6]: Missing newline:
```suggestion
end

```

File: docs/project.toml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[deps]
+Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
+Stingray = "2045e982-6adb-4e09-8364-0d2cc8a3fe4f"
```
Comment (giordano) [inline docs/project.toml:3]: This file should be called `Project.toml`, not `project.toml` (case is important).  Also, `Stingray` shouldn't be here and it'd be better to specify compat bounds (latest version of `Documenter.jl` at the moment is [`v0.27.18`](https://github.com/JuliaDocs/Documenter.jl/releases/tag/v0.27.18))
```suggestion

[compat]
Documenter = "0.27"
```

**Review by giordano — COMMENTED** [review]
Looks good overall!

**Comment by Aman-Pandey-afk** [comment]
Thanks for the awesome review! I guess the project initialization is done.

**Review by giordano — APPROVED** [review]

**Comment by matteobachetti** [comment]
Great job @Aman-Pandey-afk ! Congratulations on your first approved PR of GSoC 2022!

**Pull request merged**